lib/ecto_autoslug_field/slug_generator.ex

```diff
@@ -88,7 +88,8 @@ defmodule EctoAutoslugField.SlugGenerator do
   defp get_field_data(_, source, _) when is_binary(source), do: source
 
   defp has_value?(nil), do: false
-  defp has_value?(string) do
-    String.strip(string) != ""
+  defp has_value?(string) when is_binary(string) do
+    String.trim(string) != ""
   end
+  defp has_value?(_), do: true
```

Comment:
Could you please test this function with some corner cases?
I am not quite sure about this line, how it will behave with different types of data?

Reply:
I'll try to add some more test cases.
I just needed to slug a simple integer field, but I can see DateTime as a valid option too.

One problem is how to define if a number has a value. Is 0 a value? I suppose so.
So I'd return 'true' from this function for anything except strings.

Reply:
I added several test cases with multiple field types and they all seem to work ok.